float64[] gains
string controller_name 